tests/system/conftest.py

```diff
@@ -276,12 +295,80 @@ def intercept_stream_stream(
         return response_it
 
 
+class MetadataClientGrpcAsyncInterceptor(
+    grpc.aio.UnaryUnaryClientInterceptor,
+    grpc.aio.UnaryStreamClientInterceptor,
+    grpc.aio.StreamUnaryClientInterceptor,
+    grpc.aio.StreamStreamClientInterceptor,
+):
+    def __init__(self, key, value):
+        self._key = key
+        self._value = value
+        self.request_metadata = []
+        self.response_metadata = []
+
+    async def _add_metadata(self, client_call_details):
+        if client_call_details.metadata is not None:
+            client_call_details.metadata.append((self._key, self._value))
+            self.request_metadata = client_call_details.metadata
+
+    async def intercept_unary_unary(self, continuation, client_call_details, request):
+        await self._add_metadata(client_call_details)
+        response = await continuation(client_call_details, request)
+        metadata = [(k, str(v)) for k, v in await response.trailing_metadata()]
+        self.response_metadata = metadata
+        return response
+
+    async def intercept_unary_stream(self, continuation, client_call_details, request):
+        self._add_metadata(client_call_details)
+        response_it = continuation(client_call_details, request)
+        return response_it
+
+    async def intercept_stream_unary(
```

Comment:
We should use more descriptive names to differentiate this method from the previous one. I realize previously existing classes established this pattern, but we should clarify.